tests/metrics_test.go

```diff
@@ -153,14 +158,95 @@ var _ = Describe("Metrics", func() {
 		)
 	})
 
+	Context("SSP metrics", func() {
+		getDeployment := func(name string, namespace string) *apps.Deployment {
+			deployment := &apps.Deployment{}
+			err := apiClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, deployment)
+			Expect(err).ToNot(HaveOccurred())
+			return deployment
+		}
+
+		getOperatorPod := func() (k8sv1.Pod, uint16) {
+			pods, port := operatorPodsWithMetricsPort()
+			return pods[0], port
```

Comment:
When the operator starts, it does a leader election to determine which pod should be active. The inactive pods will not update the metrics. So the fist returned pod may be inactive, and the test would fail.

Reply:
I'm now iterating the list to make sure I get the value